### PR TITLE
[macos] Change user visible strings for new framework names

### PR DESCRIPTION
**Pull request by timrisi**

### CODE REVIEW

File: tools/mmp/driver.cs [hunk 1]
```diff
@@ -358,8 +358,8 @@ static void Main2 (string [] args)
 					true /* this is an internal option */
 				},
 				{ "xamarin-framework-directory=", "The framework directory", v => { xm_framework_dir = v; }, true },
-				{ "xamarin-full-framework", "Used with --target-framework=4.5 to select XM 4.5 Target Framework", v => { IsUnifiedFullXamMacFramework = true; } },
-				{ "xamarin-system-framework", "Used with --target-framework=4.5 to select XM 4.5 Target Framework", v => { IsUnifiedFullSystemFramework = true; } },
+				{ "xamarin-full-framework", "Used with --target-framework=4.5 to select XM Full (4.5) Target Framework", v => { IsUnifiedFullXamMacFramework = true; } },
+				{ "xamarin-system-framework", "Used with --target-framework=4.5 to select XM Full (4.5) Target Framework", v => { IsUnifiedFullSystemFramework = true; } },
```
Comment (spouliot) [inline tools/mmp/driver.cs]: I think we should remove the `(4.5)` from the descriptions, it's not correct and it's getting worse over time.

File: tools/mmp/driver.cs [hunk 2]
```diff
@@ -479,7 +479,7 @@ static void Main2 (string [] args)
 			}
 
 			if (!IsUnifiedMobile && tls_provider != null)
-				throw new MonoMacException (2011, true, "Selecting a TLS Provider is only supported in the Unified Mobile profile");
+				throw new MonoMacException (2011, true, "Selecting a TLS Provider is only supported in the Unified Modern (Mobile) profile");
```
Comment (spouliot) [inline tools/mmp/driver.cs]: That check should not be there anymore (in master) as we only support AppleTLS (removal should be done in a separate PR)

File: tools/mmp/driver.cs [hunk 3]
```diff
@@ -1177,7 +1177,7 @@ static int Compile (IEnumerable<string> internalSymbols)
 				break;
 			case "x86_64":
 				if (IsClassic)
-					throw new MonoMacException (5204, true, "Invalid architecture. x86_64 is only supported with the mobile profile.");
+					throw new MonoMacException (5204, true, "Invalid architecture. x86_64 is only supported with the Modern (Mobile) profile.");
```
Comment (spouliot) [inline tools/mmp/driver.cs]: The message (even the old one) is wrong. We don't support 64bits on classic but other XM profiles (Full or Modern) do support 64 bits.

File: tools/mmp/error.cs [hunk 4]
```diff
@@ -99,7 +99,7 @@ namespace Xamarin.Bundler {
 	//			MM52xx	linking
 	//					MM5202	Mono.framework MDK is missing. Please install the MDK for your Mono.framework version from http://mono-project.com/Downloads
 	//					MM5203  Can't find libxammac.a, likely because of a corrupted Xamarin.Mac installation. Please reinstall Xamarin.Mac.
-	//					MM5204  Invalid architecture. x86_64 is only supported with the mobile profile.
+	//					MM5204  Invalid architecture. x86_64 is only supported with the Modern (Mobile) profile.
```
Comment (spouliot) [inline tools/mmp/error.cs]: keep in sync after fix

File: tools/mmp/driver.cs (hunk 1 above)
Comment (chamons) [inline tools/mmp/driver.cs]: So I suggested keeping it to reference the old name, as we did with Modern (Mobile).

I am not stuck to the idea, and i think seeing it here it is confusing.

File: tools/mmp/driver.cs (hunk 2 above)
Comment (chamons) [inline tools/mmp/driver.cs]: Do that require us merging in the new mono first?

File: tools/mmp/driver.cs (hunk 3 above)
Comment (chamons) [inline tools/mmp/driver.cs]: Yes - Invalid architecture. x86_64 is only supported on non-Classic profiles.

File: tools/mmp/driver.cs (hunk 2 above)
Comment (spouliot) [inline tools/mmp/driver.cs]: No, that's already done (in `mmp` too IIRC) and the linker code was adjusted accordingly. Right now it's a removal of the option (not the whole code).

Comment (timrisi) [inline tools/mmp/driver.cs]: I opened another PR to remove the check and removed this change so it won't cause a merge conflict with the removal.

**Comment by chamons** [comment]
This is dont-merge until 15.2 is branched. I think it's a bit late to get QA to run through this, and it's not a high priority feature.

**Comment by timrisi** [comment]
https://bugzilla.xamarin.com/show_bug.cgi?id=53785 known failure

**Comment by spouliot** [comment]
d15-2 is branched, this is clear to go!
